tensorflow_model_optimization/g3doc/guide/clustering/clustering_example.ipynb

```diff
@@ -0,0 +1,686 @@
+{
```

Comment:
<p>Reading the available docs it's not clear what's happening inside, so I'm really left guessing.</p><p> </p><p>I assume:</p><p> </p><ul><li>The clustering tools replace weights with a smaller weight-cluster array, and indices into that array (can't that be gzipped so show the size difference directly?)</li><li><code>strip_clustering</code>  then resets to the original style of direct-weight array, but using the clustered weights? Either I don't understand: </li><li class="ql-indent-1">How this all works,</li><li class="ql-indent-1">Or why <code>strip_clustering</code>  is necessary to show the size difference here.</li></ul><p> </p><p>So maybe it could use little more explanation.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/tensorflow/model-optimization/pull/330/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

Reply:
<p>Good point - I'll make the equivalent change on the pruning side. </p><p> </p><p>I think something along the lines of "<code>strip_clustering</code>  removes any tf.Variable that clustering only needs during training, which would otherwise add to model size during serving" would work. This avoids any references to implementation details (e.g. the wrapper) </p>